### PR TITLE
Reminder email for upcoming assignments

### LINKED ISSUE #34: Configurable reminder email for upcoming assignments
From https://github.com/umts/transit-it/issues/677.

### PR DESCRIPTION
**Pull request by dfaulken**
Closes #34.

### CODE REVIEW

**Comment by dfaulken** [comment]
Also includes some paper trail schema changes, but that doesn't matter too much.

File: app/mailers/assignments_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+class AssignmentsMailer < ActionMailer::Base
+  default from: 'transit-it@admin.umass.edu'
+
+  def upcoming_reminder(assignment, user)
```
Comment (Anbranin) [inline app/mailers/assignments_mailer.rb]: Couldn't you just call this with one argument, `assignment` and then set `@user` to be `@assignment.user`?

**Comment by Anbranin** [comment]
So let's add a UI for people to add whether they want to receiver reminders and do some integration tests for that, then we cool. Looks good so far.

File: app/mailers/assignments_mailer.rb (hunk 1 above)
Comment (dfaulken) [inline app/mailers/assignments_mailer.rb]: Sure.

**Comment by dfaulken** [comment]
screen shot 2017-04-19 at 11 48 18 am

**Pull request merged**